### ISSUE TITLE
High Speed Travel Alternative Implementation

### ISSUE THREAD

**Issue by mattyx2009x** [jonafanho/Minecraft-Transit-Railway#809]
Expected Behaviour.
When aboard the plane (server) the chat should work normally and terrain should load in and when it lands the doors should open normally and allow people to exit.

Actual Behaviour.
When aboard the plane (server) the chat stops working for the time any player is on it, terrain doesn't load in at all and when the aircraft lands then nothing is loaded in and doors don't open so people can't get off. Shift to dismount also doesn't work.

Reproduction Steps.
Steps to reproduce the behaviour:

1. Create a working flight route system with the airports at least 5 thousand blocks apart or more.
2. Board the plane flying to the other destination.
3. The chat won't work so you can't message other players and all issues start here after about 15 to 30 seconds aboard the aircraft.

Crash Report.
Nothing crashed so I can't provide a log sorry.

Screenshots.
Sorry but I can't provide any also the screenshots won't help explain.

Environment.

- Minecraft Version: FORGE 1.20.1
- Installed Mods:
SCREENSHOT OF MODS FOLDER PROVIDED

Additional Information.
I am on an exaroton server. 4gb ram, problem also happens with more ram. Happens on fabric 1.19.2 server with different mods so I don't think its a mod compatibility issue (could be though). Thank you for reading this report - mattyx2009x (Minnow). For any extra information contact me via Discord, username: mattyx2009x

![Screenshot 2024-02-17 231158](https://github.com/Minecraft-Transit-Railway/Minecraft-Transit-Railway/assets/160231917/fdaa1498-ca05-49d2-8830-c5823ba38ebb)

**Comment by mattyx2009x** [comment]
Extra information.
I have added 16gb of RAM to the server and 8 CPU cores and chat does work but when I land the terrain is still not loaded. I press shift to dismount and I end up to the airport I started at. The camera is also very buggy and I can't move it nor can anyone else. Thank you for reading this extra message, remember you can always contact me for more information and screenshots via Discord (username: mattyx2009x).

**Comment by mattyx2009x** [comment]
Another bug I found is that PIDs don't work for planes anymore.

**Comment by Ultimate-USA-YT** [comment]
Having the same issue with the PIDS on my World. And my guess on the Doors not opening is Server Lag

**Comment by kettle-7** [comment]
this issue seems to be something to do with the device the files for the server are stored on, i had the same thing happen on kingsburgh when i was using an sd card for storage. my guess is the server can't read the chunks fast enough

**Comment by mattyx2009x** [comment]
yeah its probably a server issue ill try to test it on a normal world to see if it persists

**Comment by Leoparking** [comment]
飞机太快了，不是所有服务器的区块加载都能跟上。我的也是勉强能在飞行中看到地面一小块的。
如果你在飞行途中不再显示地面，并且诸如切换模式、发送聊天等服务器互动失灵，那你可能会掉线。我遇到过https://www.mtrbbs.top/thread-5314-1-1.html
实际上你掉线后重新回来发现你根本不在终点站，可能会在路途中

**Comment by leonpano2006** [comment]
https://bugs.mojang.com/secure/attachment/276318/mcbug20200326.mp4
請問你們遇到的是類似於以上影片的狀況嗎
is your issue like shown in video?

**Comment by sparrowhe** [comment]
> https://bugs.mojang.com/secure/attachment/276318/mcbug20200326.mp4 請問你們遇到的是類似於以上影片的狀況嗎 is your issue like shown in video?

Pertty like, but I can reproduced it on the MTR's train with mutiplayer server. During this bug occur, I found a lot of `moved too quickly` on my server log. And after I add [this](https://modrinth.com/mod/gotta-go-fast-fabric) mod, problem solved. That not looks like chunky load too slow, it seems like some anti-cheat feature. I'm not pretty sure about if this solution works on higher speed (i'm 80km/h now), but it works for me

**Comment by jonafanho** [comment]
For now, you can use the [Carpet](https://modrinth.com/mod/carpet) mod and enable the `antiCheatDisabled` feature. I will work on improvements with high speed travel.
